federation/pkg/kubefed/init/init.go

```diff
@@ -350,11 +374,15 @@ func createControllerManagerKubeconfigSecret(clientset *client.Clientset, namesp
 		certutil.EncodeCertPEM(entKeyPairs.controllerManager.Cert),
 	)
 
+	if dryRun {
+		return nil, nil
```

Comment:
You shouldn't do this. You should instead pass `dryRun` to `util.CreateKubeconfigSecret()`. That's what the last `false` argument to that function is here.